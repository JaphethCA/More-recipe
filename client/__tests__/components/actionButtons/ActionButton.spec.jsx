import React from 'react';
import { shallow } from 'enzyme';
import connectedActionButtons, { ActionButtons } from '../../../ActionButtons/containers/ActionButtons';
import recipeMock from '../../__mock__/recipesMock';
import mockStore from '../../__mock__/configMockStore';


const props = {
  hidden: false,
  recipe: recipeMock.recipe,
  handleUpvote: jest.fn(),
  handleDownvote: jest.fn(),
  handleAddToFavorites: jest.fn(),
  handleRemoveFromFavorites: jest.fn()
};

const store = mockStore({ favorites: recipeMock.recipes });
const state = {
  favorites: recipeMock.recipes
};


describe('<ActionButtons />', () => {
  it('renders without exploding', () => {
    const wrapper = shallow(<ActionButtons {...state} {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.length).toBe(1);
    expect(wrapper.find('ul').length).toBe(1);
    expect(wrapper.find('li').length).toBe(4);
  });

  it('should contain a <ReviewButton> element', () => {
    const wrapper = shallow(<ActionButtons {...state} {...props} />);
    expect(wrapper.find('ReviewButton').length).toBe(1);
  });

  it('should have a ".btn-list" class', () => {
    const wrapper = shallow(<ActionButtons {...state} {...props} />);
    expect(wrapper.find('ul').hasClass('btn-list')).toBe(true);
  });
  it('should match snapshots', () => {
    const wrapper = shallow(<ActionButtons {...state} {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
