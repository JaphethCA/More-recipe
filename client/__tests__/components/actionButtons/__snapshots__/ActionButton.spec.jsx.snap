// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButtons /> should match snapshots 1`] = `
<div>
  <ul
    className="btn-list"
  >
    <li>
      <ReviewButton
        hidden={false}
        reviews={1}
      />
    </li>
    <li>
      <UpvoteButton
        upvote={[Function]}
        upvotes={1}
      />
    </li>
    <li>
      <DownvoteButton
        downvote={[Function]}
        downvotes={0}
      />
    </li>
    <li>
      <FavoritesButton
        isInFavorites={[Function]}
        onFavoriteClick={[Function]}
      />
    </li>
  </ul>
</div>
`;
